source-git-monorepos/README.md

```diff
@@ -50,21 +89,367 @@ packages:
-keep things organized.
+kept in subdirectories, named according to `package_name` (the key used in
+`source-git.yaml` to reference the package), in `.distro`, to keep things
+organized.
+
+An example layout:
+
+```
+.distro
+├── backend
+│   └── copr-backend.spec
+├── cli
+│   └── copr-cli.spec
+├── frontend
+│   └── copr-frontend.spec
+└── source-git.yaml
+```
 
 This way all distribution configs and files would be kept at one place, which
-probably is somewhat better than having source-git configuration spread
-across the monorepo, in subdirectories. This setup also allows multiple paths
-from the repo mapped to a dist-git repo.
+probably is somewhat better than having source-git configuration spread across
+the monorepo, in subdirectories. This setup also allows multiple paths from
+the repo mapped to a dist-git repo.
+
+### `packit source-git init`
+
+If the source-git repository to be initialized is a monorepo, then each
+dist-git repository specified will also need to one or more subdirectories of
+the source-git repository.
+
+Which makes the whole thing a little bit too complicated ☺
+
+This complication could be elevated, if `packit source-git init` would use a
+guided (dialog-like) process, instead of taking all its inputs as arguments.
+
+As an alternative, the command could have a variant where it takes a basic
+`packages` definition to guide the initialization of a source-git monorepo.
+
+Example `packages.yaml`:
+
+```yaml
+- downstream_package_name: copr-backend
+  package_path: backend
+  dist_git_path: rpms/copr-backend
+  dist_git_branch: rawhide
+- downstream_package_name: copr-frontend
+  package_path: frontend
+  dist_git_path: rpms/copr-frontend
+  dist_git_branch: rawhide
+- downstream_package_name: copr-cli
+  package_path: cli
+  dist_git_path: rpms/copr-cli
+  dist_git_branch: rawhide
+```
+
+Which could be used as:
+
+```shell
+$ packit source-git init --packages packages.yaml v3.0.0 src/copr
+```
+
+Or the same could be read from STDIN, with `--packages=-`:
+
+```shell
+$ packit source-git init --packages=- v3.0.0 src/copr
+```
+
+Given that setting up a source-git repository is expected to be a one-time
+activity, and that there are not so many projects which are monorepos,
+implementing the above should probably have **a low priority for now**.
+
+### `packit source-git update-dist-git`
+
+**At a minimum**, the command should be updated to support monorepos, and
+transform only the content specific to the package from source-git. The
+package name (`downstream_package_name`) to be considered from
+`.distro/source-git.yaml` is the one which matches `basename DIST_GIT`.
 
-For **backwards compatibility**: if the top-level `packages` key is missing,
+**Going further**, when used with monorepos, this command could be able to
+update multiple dist-git repositories. This means, that the `DIST_GIT`
+argument of the command would become `DIST_GIT [DIST_GIT...]` (one or more
+paths to dist-git repositories).
```

Comment:
In Tito, you do `tito release` from within the upstream sub-directory.  E.g. you
do `cd backend && tito release <releaser_id>`.  Not sure what is better, but just
noting that it might be worth considering.

Reply:
Thanks for mentioning this.